### DISCUSSION TITLE
RFC: CLI redesign

### DISCUSSION THREAD

**Discussion by simonsan** [rustic-rs/rustic#987] [Ideas]
Could people give some feedback on this CLI changes?
 
The idea is to harmonize the CLI and make it easier to understand (lowering the barrier of entry), by reducing the top-level command structure to some easier to understand topics. e.g. snapshots, repository, profiles etc.

Feedback welcome! And don't worry, early WIP/design phase. 🙂 Hence your feedback is even more important! 👍🏽 

![cli_commands_categories](https://github.com/rustic-rs/rustic/assets/14062932/d5245b02-60b9-404e-9f53-c3f74e1251a5)

**Comment by kapitainsky** [comment]
Maybe it is only me but I do not think that existing CLI is particularly messy:) Coming from `restic` I feel quite familiar with CLI logic and only what I am missing is more detailed explanation of some new options.

Can it be better and more logical? I am sure yes but to achieve goal you stated:

> easier to understand (lowering the barrier of entry)

IMO it would be beneficial to focus on:

1. Documentation with examples
2. `rustic mount` - maybe using `webdav` so could work on any platform easily
3. GUI - I am thinking about something a la [kopia](https://kopia.io/) - user can create new repo/snapshot, schedule and restore.
4. Documentation with examples

and under the hood maybe regression tests and coming out of beta status:)

At the same time I have only started really using `rustic` few days ago. I will report back about any CLI shortcomings I discover.

**Reply by simonsan** [reply]
> Maybe it is only me but I do not think that existing CLI is particularly messy:)

restic's CLI structure is not necessarily messy, but it's overwhelming. And I think it's not good UX and UI, so I think it's worth it to make it better, to open up a backup tool to more non-techy people.

Your points I agree upon, though they are additional and are not a fix for bad UX/UI in terms of CLI.

That being said, I would consider us both techy people, so it's especially important that we are thinking outside the box to be more inclusive to non-techy people. We will be probably able to use any TUI/CLI. Other's aren't, when it's overwhelming, as alone a terminal can be.

**Reply by kapitainsky** [reply]
> so I think it's worth it to make it better, to open up a backup tool to more non-techy people

Not-techy people will never use CLI I am afraid.... But indeed well structured and logical CLI would be big plus for tech people.

**Reply by simonsan** [reply]
Yes, but the CLI is the foundation for the work in the repositories and commands to be reused in the GUI and TUI applications of rustic that we are designing currently. So it's a win-win situation. Give better structure to rustic and profit also for GUI and config files from the restructure.

EDIT: Moved it here, don't know why this comment became an extra comment.

**Comment by simonsan** [comment]
Feedback from Discord:

twnk — 06/01/2024 12:02
I really like this. the groups are intuitive while keeping the most important commands available easily. not a UX expert by any means, but if I was to suggest anything it would be to consider flattening maintenance (so having check, repair, prune as top level, or maybe moving prune under snapshots?). not super confident in that tho

my reasoning is when I want to check or repair I don't think "I want to repair, therefore I want to do maintenance" 

I would think "what do I want to repair? my snapshots / my index"

simonsan — 06/01/2024 12:05
i understand. i would advocate though to keep them under maintenance, as they are not used as often to have them under top level. but are useful enough at its own to be having their own outstanding category

twnk — 06/01/2024 12:06
that makes sense to me. I personally use check all the time but that's more a paranoia thing than a normal workflow I imagine...

simonsan — 06/01/2024 12:06
i'm having in mind people, that don't have the domain knowledge of restic/rustic to differentiate what could be under repository and snapshots
